hedera-node/src/main/java/com/hedera/services/context/primitives/SignedStateViewFactory.java

```diff
@@ -0,0 +1,95 @@
+package com.hedera.services.context.primitives;
+
+import com.hedera.services.ServicesState;
+import com.hedera.services.context.ImmutableStateChildren;
+import com.hedera.services.context.MutableStateChildren;
+import com.hedera.services.exceptions.NoValidSignedStateException;
+import com.hedera.services.state.migration.StateVersions;
+import com.swirlds.common.AutoCloseableWrapper;
+import com.swirlds.common.Platform;
+
+import java.util.Optional;
+
+/**
+ * Factory that provides view of last completed signed state and its state children.
+ * Fetches the latest signed state from platform for each call to get/update state children.
+ */
+public class SignedStateViewFactory {
+	private final Platform platform;
+
+	public SignedStateViewFactory(Platform platform) {
+		this.platform = platform;
+	}
+
+	/**
+	 * Try to update the mutable state children with the children from the signed state.
+	 * Throws {@code NoValidSignedStateException} when the provided state is invalid.
+	 *
+	 * @param children
+	 * 		mutable children provided
+	 * @throws NoValidSignedStateException
+	 */
+	public void tryToUpdate(MutableStateChildren children) throws NoValidSignedStateException {
+		final var state = getLatestSignedState();
+		if (!hasValidSignedState(state)) {
+			throw new NoValidSignedStateException("State children require an valid state to update");
+		}
+		children.updateFromSigned(state, state.getTimeOfLastHandledTxn());
+	}
+
+	/**
+	 * Gets the immutable state children from the latest signedState from platform. Returns {@code Optional.empty()}
+	 * when the provided state is invalid.
+	 *
+	 * @return immutable state children
+	 */
+	public Optional<ImmutableStateChildren> tryToGet() {
+		final var state = getLatestSignedState();
+		if (!hasValidSignedState(state)) {
+			return Optional.empty();
+		}
+		return Optional.of(new ImmutableStateChildren(state));
+	}
+
+	/**
+	 * Checks if the signedState provided is valid
+	 *
+	 * @param state
+	 * @return true if it is a valid signedState
+	 */
+	boolean hasValidSignedState(final ServicesState state) {
```

Comment:
I think `isValid()` will be a more idiomatic name here.

Reply:
Fixed ! Thanks !